File: src/main/java/com/nurkiewicz/tsclass/parser/ast/ClassDescriptor.java
package com.nurkiewicz.tsclass.parser.ast;

import com.google.common.collect.ImmutableList;
import lombok.Value;

@Value
public class ClassDescriptor {
    private String name;
    private ImmutableList<Field> fields;
    private ImmutableList<Method> methods;
}

File: src/main/kotlin/com/nurkiewicz/tsclass/parser/ast/ClassDescriptor.kt
package com.nurkiewicz.tsclass.parser.ast

import com.nurkiewicz.tsclass.parser.ast.expr.MethodCall

data class ClassDescriptor(
    val name: String,
    val fields: List<Field>,
    val methods: List<Method>) {

    fun findBestMatchingMethod(call: MethodCall): Method? =
            methods.find { it.matches(call) }

}

data class Field(val name: String, val type: String)
